python/tchannel/tornado/dispatch.py

```diff
@@ -54,10 +52,10 @@ def __init__(self):
         self.endpoints = {}
 
     @tornado.gen.coroutine
-    def _call_endpoint(self, endpoint, request, response):
+    def _call_endpoint(self, endpoint, request, response, proxy):
         try:
             yield tornado.gen.maybe_future(
-                endpoint.handler(request, response, endpoint.opts))
+                endpoint.handler(request, response, proxy))
         except:
             # TODO send application error
             pass
```

Comment:
You've deleted the Endpoint class. What's this `.handler` on?

Also, remove that catch-all  `except:`.